### PR TITLE
GH-5327 RDF and SPARQL 1.2 Compliant parsers, writers, SPARQL grammar and evaluation

### PR DESCRIPTION
**Pull request by desislava-hristova-ontotext**
GitHub issue resolved: #GH5327

Briefly describe the changes proposed in this PR:

The MR adds supports for RDF 1.2 and SPARQL 1.2 based on the existing branches:
GH-5327 added RDF 1.2 testsuite files for Turtle, TriG, NTriples, NQuads, and refactored testsuite code to run RDF 1.2 and 1.1 test suites GH-5327 Version Announcement in Turtle and TriG writers, Version Parsing in Turtle, TriG, NTriples, and NQuads parsers GH-5327 Changed RDF-Star to RDF 1.2 triple terms, bumped version to 6.0.0 GH-5327 RDF 1.2 implementation for JSON and dirLangString literals 
With the addition of the following:
- Triple Terms: Renamed Triple → TripleTerm throughout the codebase to align with RDF 1.2 terminology. TripleTerm is allowed only on object position as defined in RDF 1.2 specification.
- SPARQL 1.2 compliance to grammar and built-in functions. Implement SPARQL 1.2 compliant evaluation
- Add SPARQL 1.2 conformance suite. Conformance tests pass with MemoryStore. Disable SPARQL 1.2 conformance tests for NativeStore and LMDBStore since these are not implemented.
- Extends the version announcement logic so it is applied in NTriples and Nquads writers. Fixes an issue with incorrect closing of context when statement is part of the named graph and version has to be announced afterward for trig files. Fixes an issue with announcing version in the middle of ttl file where it concatenates the version to the previous triple.
- Extends RDF-XML parser and XMLLiteral serializer to fully support RDF 1.2 semantics and strict validation rules
- Updated RDF 1.2 conformance tests for parsers and writers
- Removal of SPARQL star formats -  replacement of legacy star parser/writers and formats with the corresponding classes supporting triple terms, replacement of methods and parameters indicating legacy SPARQL star support with new Triple Term support, removal of legacy star file formats (.srjs, etcs) and mime types. 
- Disabled W3CApprovedSPARQL10QueryTest - Needs further inspection
 

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

---
Co-authored-by: Sava Savov <sava.savov@graphwise.ai>
Co-authored-by: Ralitsa Dimitrova <ralitsa.dimitrova@ontotext.com>
Co-authored-by: Dancho Ivanov <dancho.ivanov@ontotext.com>

### CODE REVIEW

File: core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/ASTVar.java [hunk 1]
```diff
@@ -18,6 +18,8 @@ public class ASTVar extends SimpleNode {
 
 	private boolean anonymous;
 
+	private boolean isBNode;
```
Comment (hmottestad) [inline core/queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/ast/ASTVar.java:21]: Could you explain what this is used for?

File: core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtlestar/TurtleStarParserFactory.java
Comment (kenwenzel) [inline core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtlestar/TurtleStarParserFactory.java:1]: Do we want to keep these Turtle-star related classes at all?

Comment (desislava-hristova-ontotext) [reply core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtlestar/TurtleStarParserFactory.java:1]: No we will remove all star related classes as part of this MR. It is in the steps left to finish the MR from draft to ready.

File: core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractLiteral.java [hunk 2]
```diff
@@ -80,11 +80,12 @@ public abstract class AbstractLiteral implements Literal {
 	private static final long serialVersionUID = -1286527360744086451L;
```
Comment (hmottestad) [inline core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractLiteral.java:80]: In AbstractLiteral should the base direction be part of the equals and/or hashCode methods?

File: core/rio/api/src/main/java/org/eclipse/rdf4j/rio/RDFFormat.java [hunk 3]
```diff
@@ -196,7 +172,7 @@ public class RDFFormat extends FileFormat {
 	 * @see <a href="http://rivuli-development.com/2011/11/binary-rdf-in-sesame/">Binary RDF in Sesame</a>
 	 */
 	public static final RDFFormat BINARY = new RDFFormat("BinaryRDF", List.of("application/x-binary-rdf"), null,
-			List.of("brf"), null, SUPPORTS_NAMESPACES, SUPPORTS_CONTEXTS, SUPPORTS_RDF_STAR);
+			List.of("brf"), null, SUPPORTS_NAMESPACES, SUPPORTS_CONTEXTS, NO_TRIPLE_TERMS);
```
Comment (hmottestad) [inline core/rio/api/src/main/java/org/eclipse/rdf4j/rio/RDFFormat.java:175]: Shouldn't binary rdf support triple terms when it previously had support for RDF*?

**Comment by aschwarte10** [comment]
Over the past days I have conducted a bit of smoke testing of this change from a user perspective. I have successfully used RDF 1.2 and SPARQL 1.2 constructs in our application (in the example backed by a memory store as the first reference implementation inside RDF4J). I was able to use different serialization formats and also successfully executed different queries.

Also a quick smoke test in the RDF4J workbench was successful.

I would opt for merging this and making available as second 6.0 milestone build to a broader audience.

When there are edge cases, we can stil iterate upon those later

Great work in this PR which has been done by a broader round of contributors 👍

File: core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/TripleTermStoreTest.java
Comment (kenwenzel) [inline core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/TripleTermStoreTest.java:1]: This file seems like a copy of TripleStoreTest and needs to be removed.

**Review by kenwenzel — CHANGES_REQUESTED** [review]
Remove superfluous unrelated test in lmdb module.

File: core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/TripleTermStoreTest.java
Comment (desislava-hristova-ontotext) [reply core/sail/lmdb/src/test/java/org/eclipse/rdf4j/sail/lmdb/TripleTermStoreTest.java:1]: Right, Removed it.

**Review by aschwarte10 — APPROVED** [review]
Great work and a huge step for the RDF4J project. This PR lifts the project to the new RDF 1.2 and SPARQL 1.2 standard.

As stated in my previous comment, I have done some acceptance testing from a user's point of view. Everything working smoothly.

I would be happy to see this merged and made available in a second 6.0 milestone build for broader exposure. We can use this state as a basis for next steps (e.g. LMDB changes) and for broader validation.

Looking forward to seeing reviews from others and getting this in

**Comment by hmottestad** [comment]
Based on contributions by Sarah Grand https://github.com/SarahGrand @SarahGrand
